helix-core/src/test/java/org/apache/helix/integration/paticipant/TestCustomizedStateUpdate.java

```diff
@@ -130,52 +131,142 @@ public void testUpdateCustomizedState() {
     stateMap2.put("PREVIOUS_STATE", "STARTED");
     stateMap2.put("CURRENT_STATE", "END_OF_PUSH_RECEIVED");
     _mockProvider
-        .updateCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME, PARTITION_NAME2, stateMap2);
+        .updateCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME1, PARTITION_NAME2, stateMap2);
 
-    customizedState = _mockProvider.getCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME);
+    customizedState = _mockProvider.getCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME1);
     Assert.assertNotNull(customizedState);
-    Assert.assertEquals(customizedState.getId(), RESOURCE_NAME);
+    Assert.assertEquals(customizedState.getId(), RESOURCE_NAME1);
     mapView = customizedState.getRecord().getMapFields();
     Assert.assertEquals(mapView.keySet().size(), 2);
     Assert.assertEqualsNoOrder(mapView.keySet().toArray(),
         new String[] { PARTITION_NAME1, PARTITION_NAME2 });
 
     Map<String, String> partitionMap1 = _mockProvider
-        .getPerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME, PARTITION_NAME1);
+        .getPerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME1, PARTITION_NAME1);
     Assert.assertEquals(partitionMap1.keySet().size(), 3);
     Assert.assertEquals(partitionMap1.get("PREVIOUS_STATE"), "END_OF_PUSH_RECEIVED");
     Assert.assertEquals(partitionMap1.get("CURRENT_STATE"), "COMPLETED");
 
     Map<String, String> partitionMap2 = _mockProvider
-        .getPerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME, PARTITION_NAME2);
+        .getPerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME1, PARTITION_NAME2);
     Assert.assertEquals(partitionMap2.keySet().size(), 3);
     Assert.assertEquals(partitionMap2.get("PREVIOUS_STATE"), "STARTED");
     Assert.assertEquals(partitionMap2.get("CURRENT_STATE"), "END_OF_PUSH_RECEIVED");
 
     // test delete customized state for a partition
     _mockProvider
-        .deletePerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME, PARTITION_NAME1);
-    customizedState = _mockProvider.getCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME);
+        .deletePerPartitionCustomizedState(CUSTOMIZE_STATE_NAME, RESOURCE_NAME1, PARTITION_NAME1);
```

Comment:
did we already have deletePerPartitionCustomizedState ??

Reply:
Yes we did have deletePerPartitition.  This lineis just modified due to changing the naming of a constant. I added a new resource name constant, so the original RESOURCE_NAME --> RESOURCE_NAME1 and my new resource name == RESOURCE_NAME2

This would remove 1 partition from the customizedState map for that resource. In order to do a full reset, they'd have to delete every single partition from that map. The new API's just allow them to operate a higher level (on the resource or customizedState level, rather than on a per partition level)

Reply:
I didn't mean wrt rename. I was wondering if using existing api would have been sufficient for them? But it is ok if the existing apis are not sufficient.

Reply:
It is possible to use the deletePerPartitionCustomizedState but would require significantly more writes to be. They'd have to modify each entry in the mapFields one by one, rather than operate at the level of the znode as a whole.